File: Swift/largest-number-at-least-twice-of-others.swift
class Solution {
    func dominantIndex(_ nums: [Int]) -> Int {
        var maxNum = Int.min
        var maxIndex = -1

        for (i, num) in nums.enumerated() {
            if num >= maxNum {
                maxNum = num
                maxIndex = i
            }
        }

        for (i, num) in nums.enumerated() {
            if maxIndex != i && maxNum < num * 2 {
                return -1
            }
        }

        return maxIndex
    }
}

File: Python/largest-number-at-least-twice-of-others.py
class Solution:
    def dominantIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        maxNum = max(nums)

        for num in nums:
            if maxNum != num and maxNum < num * 2:
                return -1

        return nums.index(maxNum)
